cmd/cluster-agent/app/app.go

```diff
@@ -323,6 +323,18 @@ func start(cmd *cobra.Command, args []string) error {
 		}()
 	}
 
+	// Compliance
+	if config.Datadog.GetBool("compliance_config.enabled") {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			if err := runCompliance(mainCtx); err != nil {
```

Comment:
why don't we pass `apiCl.Cl` to `runCompliance` here? it won't be necessary to have `WaitForAPIClient`, a retrying logic has already been put in place and done in `GetAPIClient` to instantiate `apiCl` above in this function L187

Reply:
In fact `GetAPIClient` does not wait. Either you call it first and get a KubeClient or you need to call again yourself until you don't have an error anymore. And when you call it does not retry automatically, only if the backoff time isn't elapsed, leading to a `nil` `apiCl.Cl` (that's how I detected it, got panic on the first test locally without apiserver).

As I don't want to introduce a dependency between compliance and apiserver packages, I added the wait function.

Reply:
I see thanks for the explanation, I believe we could benefit from `WaitForAPIClient` to replace `GetAPIClient` at the cluster agent startup then. 

My only concern regarding not passing `apiCl.Cl` to `runCompliance` is that we're relying on the singleton more this way, and we're getting it from multiple places (DCA startup and compliance) while we could just get it once and pass it to every component at startup, I do agree about the robustness offered by `WaitForAPIClient` though, that's why I think it can replace `GetAPIClient`.

As for the dependency between compliance and the apiserver pkgs I think it's already there 😅 

This how I see it, but no strong opinions, feel free to ignore if you don't agree.

Reply:
I do agree we should replace `GetAPIClient`, though it would change a little bit the start sequence of the DCA, so needs to be validated. I think it deserves its own PR.

About not passing `apiCl.Cl`, it's also the case for `ExternalMetrics` (and maybe other places). I believe the proper way to do it is indeed to remove the singleton entirely and pass it to every component that needs it (as we can be sure to have it with `WaitForAPIClient`), but it's a significant amount of changes, so I believe it should in its own PR.

Reply:
sounds good 👍 a small TODO explaining this won't do any harm then and I'll backlog a ticket to extend the usage of `WaitForAPIClient` just like you described